### PR TITLE
WIP fix tests part deux

### PR DESCRIPTION
**Pull request by andrewhsu**

### CODE REVIEW

File: components/engine/integration-cli/docker_cli_rmi_test.go [hunk 1]
```diff
@@ -150,8 +150,8 @@ func (s *DockerSuite) TestRmiImageIDForceWithRunningContainersAndMultipleTags(c
 	dockerCmd(c, "tag", imgID, newTag)
 	runSleepingContainerInImage(c, imgID)
 
-	out, _, err := dockerCmdWithError("rmi", "-f", imgID)
-	// rmi -f should not delete image with running containers
+	out, _, err := dockerCmdWithError("rmi", imgID)
```
Comment (thaJeztah) [inline components/engine/integration-cli/docker_cli_rmi_test.go]: This is actually a bug; the change was too wide; see https://github.com/docker/cli/issues/394

(pull request to fix is here: https://github.com/docker/cli/pull/418)

File: components/engine/integration-cli/docker_cli_rmi_test.go [hunk 2]
```diff
@@ -243,7 +243,7 @@ func (s *DockerSuite) TestRmiContainerImageNotFound(c *check.C) {
 	dockerCmd(c, "rmi", "-f", imageIds[1])
 
 	// Try to remove the image of the running container and see if it fails as expected.
-	out, _, err := dockerCmdWithError("rmi", "-f", imageIds[0])
+	out, _, err := dockerCmdWithError("rmi", imageIds[0])
```
Comment (thaJeztah) [inline components/engine/integration-cli/docker_cli_rmi_test.go]: Same here

File: components/engine/integration-cli/docker_cli_swarm_test.go [hunk 3]
```diff
@@ -1730,18 +1730,13 @@ func (s *DockerSwarmSuite) TestSwarmServicePsMultipleServiceIDs(c *check.C) {
...
 	out, err = d.Cmd("service", "ps", "to")
-	c.Assert(err, checker.IsNil)
-	c.Assert(out, checker.Contains, name1+".1")
-	c.Assert(out, checker.Contains, name1+".2")
-	c.Assert(out, checker.Contains, name1+".3")
-	c.Assert(out, checker.Contains, name2+".1")
-	c.Assert(out, checker.Contains, name2+".2")
-	c.Assert(out, checker.Contains, name2+".3")
+	c.Assert(err, checker.NotNil)
+	c.Assert(out, checker.Contains, "no such service: to")
```
Comment (thaJeztah) [inline components/engine/integration-cli/docker_cli_swarm_test.go]: Alternatively, use the test I wrote at the time of the PR, but which could not be merged upstream due to not being able to update the CLI; https://github.com/moby/moby/pull/32800/commits/4242aeca230f6a0009019be195b5e0195d5ff178

That test also verifies the new behaviour

File: components/engine/client/hijack.go [hunk 4]
```diff
@@ -177,12 +177,14 @@ func (cli *Client) setupHijackConn(req *http.Request, proto string) (net.Conn, e
 
 	// Server hijacks the connection, error 'connection closed' expected
 	resp, err := clientconn.Do(req)
-	if err != nil {
-		return nil, err
-	}
-	if resp.StatusCode != http.StatusSwitchingProtocols {
-		resp.Body.Close()
-		return nil, fmt.Errorf("unable to upgrade to %s, received %d", proto, resp.StatusCode)
+	if err != httputil.ErrPersistEOF {
```
Comment (thaJeztah) [inline components/engine/client/hijack.go]: Perhaps this should be separate from the other changes; this is a back port of https://github.com/moby/moby/pull/34275/commits/7a53991bd6200b875e492d9c3f8e99a9a0c11056 (https://github.com/moby/moby/pull/34275) - "Fix panic in hijack"

**Comment by andrewhsu** [comment]
no longer needed, because 17.07 is eol

**Pull request closed without merging**